src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTests.cs

```diff
@@ -185,35 +186,18 @@ public void TestMachineName()
             Assert.NotNull(_process.MachineName);
         }
 
-        [ActiveIssue(1896)]
-        [Fact]
-        public void TestMainModule()
+        [Fact, PlatformSpecific(~PlatformID.OSX)]
+        public void TestMainModuleOnNonOSX()
         {
-            // Module support is not enabled on OSX
-            if (global::Interop.IsOSX)
-            {
-                Assert.Null(_process.MainModule);
-                Assert.Equal(0, _process.Modules.Count);
-                return;
-            }
-
-            // Ensure the process has loaded the modules.
-            Assert.True(SpinWait.SpinUntil(() =>
-            {
-                if (_process.Modules.Count > 0)
-                    return true;
-                _process.Refresh();
-                return false;
-            }, WaitInMS));
-
-            // Get MainModule property from a Process object
-            ProcessModule mainModule = _process.MainModule;
-            Assert.NotNull(mainModule);
-            Assert.Equal(CoreRunName, Path.GetFileNameWithoutExtension(mainModule.ModuleName));
+            string fileName = "CoreRun";
+            if (global::Interop.IsWindows)
+                fileName = string.Format("{0}.exe", fileName);
 
-            // Check that the mainModule is present in the modules list.
-            IEnumerable<string> processModuleNames = _process.Modules.Cast<ProcessModule>().Select(p => Path.GetFileNameWithoutExtension(p.ModuleName));
-            Assert.Contains(CoreRunName, processModuleNames);
+            Process p = Process.GetCurrentProcess();
```

Comment:
@Priya91 , @stephentoub I will be a little concerned about moving to the current process. Process tests tries to be as non-invasive as possible and hence spawns a new process for every property check, I personally think that's better.